### DISCUSSION TITLE
How should I use the tangent semantic in HLSL

### DISCUSSION THREAD

**Discussion by yillis** [MethanePowered/MethaneKit#119] [Q&A] [answered]
I'm new to this engine and I think it's great. I want to use it to implement some graphics algorithms from a paper, but when I tried to use normal mapping, I found that there is no TANGENT value in the VertexField enumeration. Is there another way to achieve this or will this feature be supported in the future? Thanks a lot!

**Comment by yillis** [comment]
I'm very sorry, I posted it in the wrong place.

**Reply by egorodet** [reply]
No problem. GitHub allows to convert any issue to discussion with one click :)

**Comment by egorodet** [comment, marked as answer]
Hello @yillis! Great to hear that you like Methane Kit framework and plan to use it for prototyping. `VertexField` enum is a part of a simple procedural Mesh generation library used in tutorial applications. Currently this "Mesh" library supports only limited use cases from tutorials and it should to be extended to loading mesh data from external file formats in future. If you want to add tangent data to the existing mesh generators in this library, you can do it in your fork and then create pull request in main repository for back merge. Alternatively, you may decide to use some external mesh loading library like [Assimp](https://github.com/assimp/assimp) and feed raw mesh data to Methane Buffers.

**Reply by yillis** [reply]
Thank you, I will try to look into the source code, and if I stick with it, I may submit a PR.
